.github/workflows/publish_tag.yml

```
@@ -107,32 +106,37 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ inputs.python-version || '3.7' }}
+          python-version: ${{ inputs.python-version }}
 
-      - id: setup-python-dependency-manager
-        name: "Install Poetry and Pipenv"
+      - name: "Setup Python Dependency Manager"
+        shell: bash
+        env:
+          INPUT_ARTIFACT_TYPE: ${{ inputs.type }}
```

We could also just completely skip this step if the input type is not 'task-script':

```suggestion
      - name: "Setup Python Dependency Manager"
        if: ${{ inputs.type === 'task-script }}
        shell: bash
```

(disclaimer: I have not tested this, but I think it would work)

---

Can we do this optimization in another ticket? Ideally we'd implement some Python dependency caching too and put the `if` statement in the above "Setup Python" step.

---

I think we'd want to also set up the python environment for IDS repos (in case they're P-IDS). So as long as this doesn't cause problems for non-Python artifacts, I'm fine keeping things the way they are for now (i.e. I agree with Vlad).